core/arch/arch.c

```diff
@@ -676,10 +677,280 @@ far_ibl_set_targets(ibl_code_t src_ibl[], ibl_code_t tgt_ibl[])
 }
 #endif
 
+/* We can't compare DR_REG_V and DR_REG_ enum values directly because they belong to
+ * different enums and we build with -Werror=enum-compare.  So we use scopes and the
+ * temporary variable reg_virtual.
+ */
+#define ASSERT_NOT_NULL_OR_INVALID(x)                                            \
+    do {                                                                         \
+        reg_id_t reg_virtual = x;                                                \
+        ASSERT((reg_virtual != DR_REG_NULL) && (reg_virtual != DR_REG_INVALID)); \
+    } while (0);
+
+/* We can't loop through DR_REG_V enums because we have a gap for value 187
+ * (== DR_REG_INVALID for x86), so we unroll the loop and compare every enum manually.
+ */
+#define CHECK_VIRTUAL_REGISTER_IDS          \
```

Comment:
Not sure we want to keep this very long sequence of check here. Perhaps we want to move it in suite/tests/api/ir_regdeps.c?

Reply:
Yes move to test.  Possible alternative is to just check the d_r_reg_virtual_names array for NULL and INVALID...it assumes that array is in synch w/ the enums.

Reply:
Moved to test. Done.